controllers/storage.go

```diff
@@ -112,17 +117,17 @@ func (s *Storage) ArtifactExist(artifact sourcev1.Artifact) bool {
 	return true
 }
 
-// Archive creates a tar.gz to the artifact path from the given dir excluding the provided file extensions
-func (s *Storage) Archive(artifact sourcev1.Artifact, dir string, excludes string, integrityCheck bool) error {
-	if excludes == "" {
-		excludes = "jpg,jpeg,gif,png,wmv,flv,tar.gz,zip"
-	}
-
+// Archive creates a tar.gz to the artifact path from the given dir excluding any VCS specific
+// files and directories, or any of the excludes defined in the excludeFiles.
+func (s *Storage) Archive(artifact sourcev1.Artifact, dir string, integrityCheck bool) error {
 	ctx, cancel := context.WithTimeout(context.Background(), s.Timeout)
 	defer cancel()
 
-	tarExcludes := fmt.Sprintf("--exclude=\\*.{%s} --exclude .git", excludes)
-	cmd := fmt.Sprintf("cd %s && tar -c %s -f - . | gzip > %s", dir, tarExcludes, artifact.Path)
+	tarExcludes := []string{"--exclude-vcs"}
```

Comment:
This change will prevent macOS users from running the controller locally. I propose we use `--exclude` with the rules from `--exclude-vcs`.